src/Swift.Runtime/src/Swift/Runtime/ProtocolWitnessTable.cs

```diff
@@ -0,0 +1,105 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Diagnostics.CodeAnalysis;
+using System.Runtime.InteropServices;
+
+namespace Swift.Runtime;
+
+public readonly struct ProtocolConformanceDescriptor : IEquatable<ProtocolConformanceDescriptor>
+{
+    private readonly IntPtr _handle;
+
+    private ProtocolConformanceDescriptor(IntPtr handle)
+    {
+        _handle = handle;
+    }
+
+    public readonly static ProtocolConformanceDescriptor Zero = default;
+
+    public bool IsValid => _handle != IntPtr.Zero;
+
+    public bool Equals(ProtocolConformanceDescriptor other)
+    {
+        return _handle == other._handle;
+    }
+
+    internal static bool TryGetProtocolConformanceDescriptor<T, U>([NotNullWhen(true)] out ProtocolConformanceDescriptor? result)
```

Comment:
Just curious - what is the scenario where this is going to be called on something which is not `ISwiftObject` ?

Reply:
csharp-only types which implement a projected Swift protocol. If we support this scenario we will need to somehow obtain the descriptor of the wrapper here. This is a future problem and we don't have a design yet in place.